<launch>

    <param name="n_pointclouds" value="3" type="int" />
    <param name="robot_base" value="base_link" type="str" />
    <param name="max_pointcloud_age" value="1" type="double" />
    <param name="publish_rate" value="10" type="int" />

    <node name="cupreds" pkg="cupreds" type="pcl_aggregator_node" output="screen">

        <remap to="/velodyne_points" from="pointcloud0" />
        <remap to="/livox/lidar" from="pointcloud1" />

    </node>

    <!-- publish the tfs -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="velodyne_broadcaster" args="0 0 0 0 0 0 1 base_link velodyne" />
    <node pkg="tf2_ros" type="static_transform_publisher" name="livox_broadcaster" args="0.09 0.17 -0.25 0 0 0 1 base_link livox_frame" />

    <!-- start the foxglove bridge -->
    <node name="foxglove_bridge" pkg="foxglove_bridge" type="foxglove_bridge" output="screen" />

    <!-- play the bag -->
    <node name="rosbag" pkg="rosbag" type="play" args="--pause --clock /catkin_ws/src/CUPREDS_ros/bags/forest.bag" output="screen" />
</launch>
